sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/ShareFileResourceTests.cs

```diff
@@ -0,0 +1,449 @@
+                .Throws(new RequestFailedException(status: 404, message: "The specified resource does not exist.", errorCode: "ResourceNotFound", default));
+            ShareFileStorageResource destinationResource = new ShareFileStorageResource(mockDestination.Object);
+
+            // Act
+            int length = 1024;
+            await TestHelper.AssertExpectedExceptionAsync<RequestFailedException>(
+                destinationResource.CopyBlockFromUriInternalAsync(sourceResource, new HttpRange(0, length), false, length),
+                e =>
+                {
+                    Assert.AreEqual("ResourceNotFound", e.ErrorCode);
+                });
+        }
+
+        [Test]
+        public async Task GetPropertiesAsync()
+        {
+            // Arrange
+            Mock<ShareFileClient> mock = new(
+                new Uri("https://storageaccount.file.core.windows.net/container/file"),
+                new ShareClientOptions());
+
+            long length = 1024;
+            mock.Setup(b => b.GetPropertiesAsync(It.IsAny<ShareFileRequestConditions>(), It.IsAny<CancellationToken>()))
+                .Returns(Task.FromResult(Response.FromValue(
+                    FilesModelFactory.StorageFileProperties(
+                        lastModified: DateTime.MinValue,
+                        metadata: default,
+                        contentLength: length,
+                        contentType: default,
+                        eTag: new ETag("etag"),
+                        contentHash: default,
+                        contentEncoding: default,
+                        cacheControl: default,
+                        contentDisposition: default,
+                        contentLanguage: default,
+                        copyCompletedOn: DateTimeOffset.MinValue,
+                        copyStatusDescription: default,
+                        copyId: default,
+                        copyProgress: default,
+                        copySource: "https://storageaccount.file.core.windows.net/container/file2",
+                        copyStatus: CopyStatus.Success,
+                        isServerEncrypted: false,
+                        fileAttributes: default,
+                        fileCreationTime: DateTimeOffset.MinValue,
+                        fileLastWriteTime: DateTimeOffset.MinValue,
+                        fileChangeTime: DateTimeOffset.MinValue,
+                        filePermissionKey: default,
+                        fileId: default,
+                        fileParentId: default),
+                    new MockResponse(200))));
+
+            ShareFileStorageResource storageResource = new ShareFileStorageResource(mock.Object);
+
+            // Act
+            StorageResourceProperties result = await storageResource.GetPropertiesInternalAsync();
+
+            // Assert
+            Assert.NotNull(result);
+        }
```

Comment:
Should we be asserting that the expected properties that translate over to the new model actually do translate over?

Reply:
Since the `StorageResourceProperties` members are all internal, there's not a way to access them.

I did think of sharing the properties class in the test package but I think we would run into a double dependency compile issue.

At the end of the day we are going to remove the `StorageResourceProperties` call and potentially the `GetProperties` call with it.

Unless you have a suggestion on how to grab the values internally. Open to suggestions here on how to do this.